### ISSUE TITLE
Problem with ng-repeat

### ISSUE THREAD

**Issue by tiriana** [benjamincharity/angular-flickity#38]
Hello @benjamincharity 

When I don't use ng-repeat - just plain HTML - everything works like a charm. 
But I have a problem when using ```ng-repeat``` with ```bc-flickity``` directive. It seems it runs after ```bc-flickity``` ```postLink``` is finished.
I put a breakpoint in this line https://github.com/benjamincharity/angular-flickity/blob/master/src/flickity.directive.js#L53 to find out that:

1. When using plan html (listing all slides literally) - ```element[0]``` contains all slides DOM elements. As expected, nothing fancy here.
2. But when I use ```ng-repeat``` - ```element[0]``` is empty. 

I saw one of yours examples: http://plnkr.co/edit/qdyKKo?p=preview
and I implemented it in my code, and it does not work either for me. Which brings me to conclusion that there might be something wrong with my code. 
The problem is that this is a code that was handed over to me, and I'm not really sure what can cause this issue.

This might not be a bug of your code at all, but I'd appreciate any help/hints from you.

**Comment by benjamincharity** [comment]
Hey @tiriana, sorry you're hitting an issue here. `ng-repeat` definitely *should* work. (honestly never even tried it without `ng-repeat` and I'm using it in a few different projects now). Of course, I'll never rule out an issue with the library itself.

I would be more than happy to look into it if you can reproduce the issue somewhere that I can see/touch it. Maybe a Plunker/JSFiddle/CodePen etc.

Even if it's something related specifically to your code, I'm sure someone else will run into the same issue.

**Comment by tiriana** [comment]
@benjamincharity I'll try to reproduce it. 

Breaking news: I edited sources locally, just to check. And after wrapping https://github.com/benjamincharity/angular-flickity/blob/master/src/flickity.directive.js lines from 53 to 60 info ```setTimeout(function (){...}, 0 )``` everything looks like a charm. ugly-but-works

**Comment by benjamincharity** [comment]
Very interesting.. curious as to why this fixes it.. I've definitely had to use Angular's `$timeout` before to keep 3rd party libs in sync with Angular's cycles. I don't believe using a native `setTimeout` has the same effect, so it seems this may be more DOM related than Angular digest cycle related..

**Comment by tiriana** [comment]
```$timeout``` also helps.
If I'll find  away to reproduce it I will let you know.

**Comment by benjamincharity** [comment]
Ahh maybe Angular is tying standard timeouts into the digest cycle.. I'll definitely have to look more into that.

**Comment by benjamincharity** [comment]
Hey @tiriana, are you by any chance fetching your slider data from a remote source? I have seen an error similar to this in relation to fetching data via ajax. Wrapping the service call to `create()` in a `$timeout` fixed that issue much like you reported here.

Here is a demo using a `$timeout` with `create()`: http://plnkr.co/edit/hkT7Tx?p=preview

Closing for now since the solution you found seems to match what I have found. Feel free to continue the discussion if you are still having issues though!
